sapl-spring-data-mongo-reactive/src/test/java/io/sapl/springdatamongoreactive/sapl/queries/enforcement/ProceededDataFilterEnforcementPointTest.java

```diff
@@ -88,7 +87,7 @@ void when_actionWasFoundInPolicies_then_enforce() throws JsonProcessingException
         MockedConstruction<DataManipulationHandler> dataManipulationHandlerMockedConstruction = mockConstruction(
                 DataManipulationHandler.class)) {
             // GIVEN
-            var obligations     = objectMapper.readTree(
+            var obligations     = (ArrayNode) objectMapper.readTree(
                     "[{\"type\":\"mongoQueryManipulation\",\"conditions\":[\"{'role':  {'$in': ['USER']}}\"]},{\"type\":\"filterJsonContent\",\"actions\":[{\"type\":\"blacken\",\"path\":\"$.firstname\",\"discloseLeft\":2}]},{\"type\":\"jsonContentFilterPredicate\",\"conditions\":[{\"type\":\"==\",\"path\":\"$.id\",\"value\":\"a1\"}]}]");
```

Comment:
Also, I recommend replacing this with a multi-line string literal as supported by modern JDKs. That eliminates the need for escaping " all the time and is much more readable,

Reply:
check everywhere I will not repeat this at each location in the review.